File: python/14.py
def get_next(n):
    if (n%2 == 0):
        return n/2
    else:
        return (n*3)+1

def get_chain(start):
    chain = []
    n = start
    while (n != 1):
        chain.append(n)
        n = get_next(n)
    return chain


longest = 0

for i in range(1,1000001):
    chain = get_chain(i)
    length = len(chain)
    if (length > longest):
        longest = length
        print("%i, length: %i" % (i, length))

File: scala/14.scala
object Euler14
{
    def get_chain(n: Long, c: Int = 0): Int = 
	{
		if (n == 1) 
		{
			return c + 1
		}
		else
		{
			if(n % 2 == 0) 
			{
				return get_chain(n / 2, c+1)
			}
			else
			{
				return get_chain(3 * n + 1, c+1)
			}
		} 
	}

    def main(args: Array[String])
    {
        var longest = 0

        for (i <- (1 to 1000000))
        {
        	val length = get_chain(i)

	        if (length > longest)
	        {
	        	longest = length
	            println(i + ", length: " + length)
	        }
        }

    }
}
